test/validation/mod2c_core/cpp/NapDA.cpp

```diff
@@ -620,10 +623,7 @@ for (;;) { /* help clang-format properly indent */
  v=_v;
 {
  {  
-  #if !defined(_derivimplicit_states_NapDA)
-    #define _derivimplicit_states_NapDA 0
-  #endif
-  derivimplicit_thread(2, _slist1, _dlist1, _derivimplicit_states_NapDA, _threadargs_);
+  states_NapDA(_threadargs_);
```

Comment:
@nrnhines: just want to bring this to your attention: as you might remember very well, OpenACC/GPU execution didn't support function pointers and hence we needed to implement [kinderiv.py](https://github.com/BlueBrain/CoreNeuron/blob/423ae6c30491d01107bfe0efe18023bb7dacc015/coreneuron/kinderiv.py) to avoid the same.

In this PR Olli Is changing code generation slightly to create a functor where the `operator()` now encapsulates the function we want to call on GPU i.e.

```cpp
struct foo {
  int operator()() const;
};

int foo::operator()() const { ... }
void bar() {
  #pragma acc ...
  for(...) {
    solver(foo{});
  }
}
```

With this, `foo{}` can be copied onto GPU, and hence the GPU version of the function is indirectly accessible. As you can see in https://github.com/BlueBrain/CoreNeuron/pull/809, this helps to remove all `kinderiv.py` related parts. Quite an improvement!

Reply:
It's great that Olli could drain the swamp!